dbms/src/Columns/ColumnDecimal.cpp

```diff
@@ -117,6 +121,36 @@ MutableColumnPtr ColumnDecimal<T>::cloneResized(size_t size) const
     return std::move(res);
 }
 
+template <typename T>
+void ColumnDecimal<T>::insertData(const char * pos, size_t /*length*/)
+{
+    data.push_back(T());
+
+    const void * src = pos;
+    void * dst = &data.back();
+
+#if __SSE2__
+    /// prevent aligned SSE load/store.
+    if constexpr (sizeof(T) == 4)
+    {
+        __m128 value = _mm_load_ss(static_cast<const float *>(src));
+        _mm_store_ss(static_cast<float *>(dst), value);
+    }
+    else if constexpr (sizeof(T) == 8)
+    {
+        __m128d value = _mm_load_sd(static_cast<const double *>(src));
+        _mm_store_sd(static_cast<double *>(dst), value);
+    }
+    else if constexpr (sizeof(T) == 16)
+    {
+        __m128i value = _mm_loadu_si128(static_cast<const __m128i *>(src));
+        _mm_storeu_si128(static_cast<__m128i *>(dst), value);
+    }
+#else
+    memcpy(dst, src, sizeof(T));
```

Comment:
This will work in every case.

Reply:
Yup. I have removes all SSE stuff. It's probably better to make tiny memcpy from library inlined function and use it.